packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1034,10 +1034,57 @@ class PopupMenuButton<T> extends StatefulWidget {
   final bool captureInheritedThemes;
 
   @override
-  _PopupMenuButtonState<T> createState() => _PopupMenuButtonState<T>();
+  PopupMenuButtonState<T> createState() => PopupMenuButtonState<T>();
 }
 
-class _PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
+/// The [State] for a [PopupMenuButton].
+///
+/// It allows you to call [showButtonMenu] manually, which is normally
+/// called when the user taps the button.
+///
+/// You can obtain this state by providing a [GlobalKey] to your
+/// [PopupMenuButton], i.e. `GlobalKey<PopupMenuButtonState>` to
+/// the `key` parameter.
+///
+/// {@tool sample}
+///
+/// You would access your [PopupMenuButtonState] using a [GlobalKey].
+/// Using the following setup, you can show the menu
+/// using `globalKey.currentState.showButtonMenu`.
+///
+/// ```dart
+/// class ExampleWidget extends StatelessWidget {
+///   final GlobalKey<PopupMenuButtonState<int>> globalKey = GlobalKey();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return PopupMenuButton<int>(
+///       key: globalKey,
+///       itemBuilder: (BuildContext context) {
+///         return <PopupMenuEntry<int>>[
+///           const PopupMenuItem<int>(
+///             value: 1,
+///             child: Text('Tap me please!'),
+///           ),
+///         ];
+///       },
+///     );
+///   }
+///
+///   void showButtonMenu() => globalKey.currentState.showButtonMenu();
+/// }
+/// ```
+///
+/// {@end-tool}
+///
+/// Alternatively, you could also visit child elements of your parent widget to obtain
+/// the state if you know exactly that the widget is a direct descendant.
+class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
+  /// This will show a popup menu with the buttons supplied to
+  /// [PopupMenuButton.itemBuilder] at the position of your [PopupMenuButton].
+  ///
+  /// This is called by the state itself when the user taps the button
+  /// and [PopupMenuButton.enabled] is set to `true`.
```

Comment:
we also mention that developer call this method to show the menu programatically.